library/src/main/java/com/stetcho/rxwifip2pmanager/data/wifi/RxWifiP2pManager.java

```diff
@@ -220,7 +228,7 @@ public void onFailure(int reason) {
      * @return a {@link Single} observable that emits the intent which indicated that p2p peers
      * changed
      */
-    private Single<Intent> listenForNewPeers(Single<Void> voidSingle) {
+    private Single<Intent> listenForNewPeers(Single<Object> voidSingle) {
```

Comment:
Might be affected by the above comment regarding converting this to a `Completable`. Perhaps it might make sense to convert this to an `ObservableTransformer`.